### PR TITLE
[CARBONDATA-3544] cli support summary statistics for all columns

### PR DESCRIPTION
**Pull request by QiangCai**
add option -C to show statistics for all columns

for example:
```
## Statistics for All Columns
Shard #1 (211120282868026_batchno0-0-null-211119172480223)
Block  Blocklet  Column Name  Meta Size  Data Size  
0      0         name         1.90KB     2.15KB     
0      0         age          3.36KB     2.57MB     
0      1         name         1.90KB     2.16KB     
0      1         age          3.36KB     2.57MB     
1      0         name         1.90KB     2.16KB     
1      0         age          3.36KB     2.57MB     
1      1         name         1.90KB     2.16KB     
1      1         age          3.36KB     2.57MB     
2      0         name         1.90KB     2.14KB     
2      0         age          3.36KB     2.57MB     
2      1         name         1.18KB     1.33KB     
2      1         age          2.04KB     1.49MB     

Shard #2 (211128474061207_batchno0-0-null-211128385497539)
Block  Blocklet  Column Name  Meta Size  Data Size  
0      0         name         1.90KB     2.15KB     
0      0         age          3.36KB     2.57MB     
0      1         name         1.90KB     2.16KB     
0      1         age          3.36KB     2.57MB     
1      0         name         1.90KB     2.16KB     
1      0         age          3.36KB     2.57MB     
1      1         name         1.90KB     2.16KB     
1      1         age          3.36KB     2.57MB     
2      0         name         1.90KB     2.14KB     
2      0         age          3.36KB     2.57MB     
2      1         name         1.18KB     1.33KB     
2      1         age          2.04KB     1.49MB
```

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
   no, just add a new option
 - [x] Any backward compatibility impacted?
   no
 - [x] Document update required?
   need
 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
            added
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
              no
        - Any additional information to help reviewers in testing this change.
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
 small changes

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/631/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/636/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/636/

File: tools/cli/src/test/java/org/apache/carbondata/tool/CarbonCliTest.java [hunk 1]
```diff
@@ -260,6 +264,31 @@ public void testSummaryPageMeta() {
...
   }
 
+  @Test
+  public void testSummaryAllColumns() {
+    String[] args = { "-cmd", "summary", "-p", path, "-C" };
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    PrintStream stream = new PrintStream(out);
+    CarbonCli.run(args, stream);
+    String output = new String(out.toByteArray());
+    System.out.println(output);
```
Comment (xuchuanyin) [inline tools/cli/src/test/java/org/apache/carbondata/tool/CarbonCliTest.java]: avoid directly printing to console, use logger instead

File: tools/cli/src/test/java/org/apache/carbondata/tool/CarbonCliTest.java [hunk 2]
```diff
@@ -260,6 +264,31 @@ public void testSummaryPageMeta() {
...
+  public void testSummaryAllColumnsForOneFile() {
+    CarbonFile folder = FileFactory.getCarbonFile(path);
+    CarbonFile[] carbonFiles =
+        folder.listFiles(file -> file.getName().endsWith(CarbonTablePath.CARBON_DATA_EXT));
+    String[] args = { "-cmd", "summary", "-p", carbonFiles[0].getCanonicalPath(), "-C" };
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    PrintStream stream = new PrintStream(out);
+    CarbonCli.run(args, stream);
+    String output = new String(out.toByteArray());
+    System.out.println(output);
```
Comment (xuchuanyin) [inline tools/cli/src/test/java/org/apache/carbondata/tool/CarbonCliTest.java]: avoid directly printing to console, use logger instead

**Comment by xuchuanyin** [comment]
@QiangCai forget to update the doc as well?

**Comment by QiangCai** [comment]
@xuchuanyin carboncli has no doc...

File: tools/cli/src/main/java/org/apache/carbondata/tool/DataSummary.java [hunk 3]
```diff
@@ -371,6 +373,32 @@ private void printColumnStats(String columnName) throws IOException, MemoryExcep
     printer.collectFormattedData();
   }
 
+  private void printAllColumnStats() {
+    if (!dataFiles.isEmpty()) {
+      outPuts.add("");
```
[the remark this answers is not in the document]
Comment (QiangCai) [reply tools/cli/src/main/java/org/apache/carbondata/tool/DataSummary.java]: add an empty line to separate the output information

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonCliCommand.scala [hunk 4]
```diff
@@ -49,12 +49,22 @@ case class CarbonCliCommand(
...
-    val commandArgs: Seq[String] = commandOptions.split("\\s+")
-    val finalCommands = commandArgs.collect {
-      case a if a.trim.equalsIgnoreCase("summary") || a.trim.equalsIgnoreCase("benchmark") =>
-        Seq(a, "-p", carbonTable.getTablePath)
-      case x => Seq(x.trim)
-    }.flatten
+    val commandArgs: Seq[String] = commandOptions.split("\\s+").map(_.trim)
+    commandArgs.exists { comand =>
+      comand.equalsIgnoreCase("-p")
+    }
```
[the remark this answers is not in the document]
Comment (QiangCai) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonCliCommand.scala]: fixed

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonCliCommand.scala [hunk 5]
```diff
@@ -49,12 +49,22 @@ case class CarbonCliCommand(
...
+    } else {
+      val needPath = commandArgs.exists { command =>
+        command.equalsIgnoreCase("summary") || command.equalsIgnoreCase("benchmark")
+      }
+      if (needPath) {
+        commandArgs ++ Seq("-p", carbonTable.getTablePath)
+      } else {
+        commandArgs
+      }
+    }
```
[the remark this answers is not in the document]
Comment (QiangCai) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonCliCommand.scala]: done

File: tools/cli/src/test/java/org/apache/carbondata/tool/CarbonCliTest.java (hunk 1 above)
Comment (QiangCai) [reply tools/cli/src/test/java/org/apache/carbondata/tool/CarbonCliTest.java]: removed

File: tools/cli/src/test/java/org/apache/carbondata/tool/CarbonCliTest.java (hunk 2 above)
Comment (QiangCai) [reply tools/cli/src/test/java/org/apache/carbondata/tool/CarbonCliTest.java]: removed

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/634/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/639/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/639/

File: tools/cli/src/main/java/org/apache/carbondata/tool/CarbonCli.java [hunk 6]
```diff
@@ -79,6 +79,9 @@ private static Options buildOptions() {
         .withDescription("column to print statistics")
         .withLongOpt("column")
         .create("c");
+    Option columns = OptionBuilder
+        .withDescription("print statistics for all columns")
+        .create("C");
```
Comment (jackylk) [inline tools/cli/src/main/java/org/apache/carbondata/tool/CarbonCli.java]: add a `.withLongOpt("columns")`

Comment (QiangCai) [reply tools/cli/src/main/java/org/apache/carbondata/tool/CarbonCli.java]: done

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/636/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/641/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/641/

**Comment by jackylk** [comment]
LGTM

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**